automation/version/version_file.py

```diff
@@ -49,6 +51,23 @@ def create_or_update_version_file(mlrun_version):
     except Exception as exc:
         logger.warning("Failed to get version", exc_info=exc)
 
+    # Check if the provided version is a semver and followed by a "-"
+    semver_pattern = r"^[0-9]+\.[0-9]+\.[0-9]+"  # e.g. 0.6.0-
+    rc_semver_pattern = rf"{semver_pattern}-(a|b|rc)[0-9]+$"
+
+    # In case of semver - do nothing
+    if re.match(semver_pattern, mlrun_version):
+        pass
+
+    # In case of semver - replace the first occurrence of "-" with "+" to align with PEP 440
+    # https://peps.python.org/pep-0440/
```

Comment:
```suggestion

    # In case of rc semver - replace the first occurrence of "-" with "+" to align with PEP 440
    # https://peps.python.org/pep-0440/
```